### PR TITLE
RPP Tensor Audio Support - Down Mixing

### PR DESCRIPTION
**Pull request by r-abishek**
- Adds support for Audio functionality - Down Mixing
- Adds relevant QA and performance tests

### CODE REVIEW

File: src/include/cpu/rpp_cpu_simd.hpp [hunk 1]
```diff
@@ -2438,6 +2438,29 @@ static inline __m128 log_ps(__m128 x)
     return x;
 }
 
+inline Rpp32f rpp_hsum_ps(__m128 x)
+{
+    __m128 shuf = _mm_movehdup_ps(x);        // broadcast elements 3,1 to 2,0
+    __m128 sums = _mm_add_ps(x, shuf);
```
Comment (rrawther) [inline src/include/cpu/rpp_cpu_simd.hpp:2444]: why aren't we using built-in hadd instructions?

File: src/modules/cpu/kernel/down_mixing.hpp [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+*/
+
+#include "rppdefs.h"
+#include <omp.h>
+
+RppStatus down_mixing_host_tensor(Rpp32f *srcPtr,
+                                  RpptDescPtr srcDescPtr,
+                                  Rpp32f *dstPtr,
+                                  RpptDescPtr dstDescPtr,
+                                  Rpp32s *srcLengthTensor,
```
Comment (rrawther) [inline src/modules/cpu/kernel/down_mixing.hpp:32]: It is better to combine both channels and Length into a single tensor

**Review by rrawther — CHANGES_REQUESTED** [review]
please address review comments

File: src/include/cpu/rpp_cpu_simd.hpp [hunk 3]
```diff
@@ -2438,6 +2438,29 @@ static inline __m128 log_ps(__m128 x)
     return x;
 }
 
+inline Rpp32f rpp_hsum_ps(__m128 x)
+{
```
Comment (LakshmiKumar23) [inline src/include/cpu/rpp_cpu_simd.hpp:2442]: @r-abishek Are we not following Google style coding for RPP? With all opening { on the same line as function name

File: src/include/cpu/rpp_cpu_simd.hpp (hunk 1 above)
Comment (sampath1117) [reply src/include/cpu/rpp_cpu_simd.hpp:2444]: Hi Rajy,
With the usage of built-in  _mm_hadd_ps, the performance was slower,  so we used the above alternate method for doing sum of all values in __m128 register. The above approach is nearly 20% faster than the approach with built-in instruction

File: src/include/cpu/rpp_cpu_simd.hpp (hunk 3 above)
Comment (r-abishek) [reply src/include/cpu/rpp_cpu_simd.hpp:2442]: Once the BatchPD->Tensor migration is complete and we are in a position to phase out BatchPD code, there will be a lot less code to deal with, and a single PR to format the whole repo could be done to adhere to google std.
For now, there's easily a 90+% majority of code in RPP in this format, so retaining this

File: src/include/cpu/rpp_cpu_simd.hpp (hunk 1 above)
Comment (rrawther) [reply src/include/cpu/rpp_cpu_simd.hpp:2444]: Can you please make a sample code with both ways. We need to run it on different CPU versions

Comment (sampath1117) [reply src/include/cpu/rpp_cpu_simd.hpp:2444]: H Rajy,
Adding a snippet of sample for both variations for your reference
image

In Intel intrinsic guide, _mm_hadd_ps latency is available only for Skylake 
On Skylake, Built-in variant takes a total latency of 19 cycles whereas custom variant takes 14 cycles

File: src/modules/rppt_tensor_audio_augmentations.cpp [hunk 4]
```diff
@@ -126,3 +126,31 @@ RppStatus rppt_pre_emphasis_filter_host(RppPtr_t srcPtr,
...
+                                dstDescPtr,
+                                srcDimsTensor,
+                                normalizeWeights,
+                                rpp::deref(rppHandle));
+
+        return RPP_SUCCESS;
+    }
+    else
+    {
+        return RPP_ERROR_NOT_IMPLEMENTED;
```
Comment (rrawther) [inline src/modules/rppt_tensor_audio_augmentations.cpp:154]: Don't we need to have FP16 as well?

**Review by rrawther — COMMENTED** [review]
Added comment

File: src/modules/rppt_tensor_audio_augmentations.cpp (hunk 4 above)
Comment (r-abishek) [reply src/modules/rppt_tensor_audio_augmentations.cpp:154]: All 9 audio functions are currently in F32. Especially with functions using Decibels, small precision changes are affecting final outputs or training accuracy. 
We can possibly do a F16-study on porting all these once we have all 9 in F32 in rpp ToT.

**Comment by r-abishek** [comment]
@rrawther This PR has tests passing and comments addressed. Could be reviewed

**Review by rrawther — APPROVED** [review]

**Pull request merged**